File: 653-two-sum-iv-input-is-a-bst/653-two-sum-iv-input-is-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    void vadd(vector<int> & v,TreeNode* root)
    {
        if(root == NULL)
        {
            return;
        }
        
        
        vadd(v,root->left);
        v.push_back(root->val);
        vadd(v,root->right);
        
        return ;
    }
    
    bool findTarget(TreeNode* root, int k) {
        
        vector<int> v;
        
        vadd(v,root);
       
        int i = 0;
        int j = v.size()-1;
        int sum;
        
        while(i<j)
        {
            sum = v[i]+v[j];
            

            if(sum == k)
            {
                return true;
            }
            else if(sum > k)
            {
                sum -= v[j];
                j--;
            }
            else if(sum < k)
            {
                sum -= v[i];
                i++;
            }
        }
        
        
        return false;
    }
};

File: 653-two-sum-iv-input-is-a-bst/653-two-sum-iv-input-is-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorder(self,root,arr):
        if not root:
            return
        self.inorder(root.left,arr)
        arr.append(root.val)
        self.inorder(root.right,arr)
        return arr
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        arr=[]
        
        result = self.inorder(root,arr)
        # print(result,k)
        for i in range(len(result)):
            for j in range(len(result)):
                if (result[i] + result[j] ==k) & (i!=j) :
                    # print(result[i] + result[j],i,j)
                    return True
        return False

        